metsrw/mets.py

```diff
@@ -303,25 +303,39 @@ def serialize(self, fully_qualified=True):
         root.append(self._normative_structmap())
         return root
 
-    def tostring(self, fully_qualified=True, pretty_print=True):
+    def tostring(self, fully_qualified=True, pretty_print=True, encoding="UTF-8"):
         """
         Serialize and return a string of this METS document.
 
-        To write to file, see :meth:`write`
+        To write to file, see :meth:`write`.
+
+        The default encoding is ``UTF-8``. This method will return a unicode
+        string when ``encoding`` is set to ``unicode``.
```

Comment:
This is clearly documented but it seems non obvious — unicode isn't an encoding, utf-8 is an encoding. Can we just make it always return bytes? (Although that is also problematic, because the method is called `tostring`, but that's what lxml does at least). Or, does it make more sense just to pass an `xml_declaration` kwarg through?

Reply:
I thought so too but `"unicode"` is an accepted value by lxml's `encoding` kwarg.
They actually mention it in their tutoriall too: https://lxml.de/tutorial.html.

![image](https://user-images.githubusercontent.com/606459/55248959-2b3aca80-5208-11e9-86c0-7ac452ccdd74.png)

How would the signature of the method would look like?

Reply:
Doh! Ok, never mind then 😁 I think you've done it correctly, since we're aping lxml's `tostring` method here, we should keep the same behavior.